### PR TITLE
Magicseaweed sensor documentation

### PR DESCRIPTION
**Pull request by jcconnell**
**Description:**

Documentation for the magicseaweed sensor.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#15132

## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

### CODE REVIEW

File: source/_components/sensor.magicseaweed.markdown [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+description: "How to integrate Magicseaweed within Home Assistant."
+date: 2018-06-24 21:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: magicseaweed.png
+ha_category: Sensor
+featured: false
+ha_release: "0.72"
```
Comment (frenck) [inline source/_components/sensor.magicseaweed.markdown]: 0.72 is already out. You might want to up this to 0.73 :arrow_up:

File: source/_components/sensor.magicseaweed.markdown [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      - 12PM
+      - 3PM
+      - 6PM
+    monitored_conditions:
+      - swell_forecast
+      - min_breaking_swell
+      - max_breaking_swell
+```
+
+Configuration variables:
```
Comment (frenck) [inline source/_components/sensor.magicseaweed.markdown]: Please use the configuration tags, for more information please see:
https://home-assistant.io/developers/documentation/create_page/#configuration

File: source/_components/sensor.magicseaweed.markdown [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+You need an API key which is free but requires [registration](https://magicseaweed.com/developer/sign-up). Magicseaweed is limiting users of the API to 2 spots for the free plan.
+
+To add Magicseaweed forecasts to your installation, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+sensor:
+  - platform: magicseaweed
+    name: Hurricane Spot
```
Comment (frenck) [inline source/_components/sensor.magicseaweed.markdown]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

File: source/_components/sensor.magicseaweed.markdown [hunk 4]
```diff
@@ -0,0 +1,66 @@
+---
+layout: page
+title: "Magicseaweed Sensor"
+description: "How to integrate Magicseaweed within Home Assistant."
+date: 2018-06-24 21:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: magicseaweed.png
```
Comment (frenck) [inline source/_components/sensor.magicseaweed.markdown:10]: This logo is missing in the repository, maybe you forgot to add it?

**Review by frenck — DISMISSED** [review]
Hi @jcconnell,

I've reviewed your documentation PR. Could you take a look at the comments? 👍

File: source/_components/sensor.magicseaweed.markdown (hunk 4 above)
Comment (jcconnell) [reply source/_components/sensor.magicseaweed.markdown:10]: Yes, I did forget to add it. I'm still new at this. What's the procedure for adding it / where does it go?

Comment (jcconnell) [reply source/_components/sensor.magicseaweed.markdown:10]: Just added. A transparent background image was not available. According to their API documentation, they request that the logo I uploaded be used.

**Comment by frenck** [comment]
@jcconnell  Thanks for the updates, nevertheless, the builds are failing. 
Could you take a look? Thx! 👍

**Comment by jcconnell** [comment]
Hello @frenck, I apologize for the delay. I was asked to write a library to supplement this sensor. Now that it's finished, I'm revisiting this error with the documentation. Jekyll is reporting an error on line 9 column 25 regarding mapping values on sensor.magicseaweed.markdown. I've reviewed the file and searched for similar problems, but I'm still unable to find the issue. Do you see anything I might be missing?

**Comment by jcconnell** [comment]
Ok, it seems that I have addressed all the problems with my file, sensor.magicseaweed.markdown. It is now complaining about a filename for MQTT logging:

`Invalid filename 'components/mqtt/#logging.html'. Deployed filenames cannot contain # or ? characters`

**Comment by frenck** [comment]
@jcconnell  Rebase this PR onto `origin/next`, that will probably do the trick.

**Comment by jcconnell** [comment]
Sorry, I'm not very familiar with this workflow. What would be the process to rebase this onto origin/next? Can it be completed through a web browser?

**Comment by ericnewton76** [comment]
@jcconnell i'll walk you through this via teamviewer

**Comment by MimbaMonkeyHouse** [comment]
Did anyone have any luck with this? Tell us if help is needed...

File: source/_components/sensor.magicseaweed.markdown [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    3PM:
+      description: Display forecast for 3PM.
+    6PM:
+      description: Display forecast for 6PM.
+    9PM:
+      description: Display forecast for 9PM.
+    12AM:
+      description: Display forecast for 12AM.
+spot_id:
+  description: Surf spot ID to monitor surf forecast of.Details for getting spot id available at [Magicseaweed](https://magicseaweed.com/developer/forecast-api)
```
Comment (MartinHjelmare) [inline source/_components/sensor.magicseaweed.markdown]: There's a missing space after the period before "Details". But maybe rewrite the description somewhat? It reads a bit weird when ending with the preposition in the first sentence.

File: source/_components/sensor.magicseaweed.markdown [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  required: true
+  type: string
+monitored_conditions:
+  description: Conditions to display in the frontend.
```
Comment (MartinHjelmare) [inline source/_components/sensor.magicseaweed.markdown]: This isn't dependent on the frontend per se, but decides what entities to create, ie what data to make available.

**Review by MartinHjelmare — APPROVED** [review]

**Comment by MartinHjelmare** [comment]
Can be merged when build passes.

**Comment by jcconnell** [comment]
Yes! Thanks everyone for the help and patience.

**Pull request merged**